compiler/rustc_ast_lowering/src/item.rs

```diff
@@ -245,9 +248,30 @@ impl<'hir> LoweringContext<'_, 'hir> {
             },
             ItemKind::ForeignMod(fm) => hir::ItemKind::ForeignMod {
                 abi: fm.abi.map_or(abi::Abi::FALLBACK, |abi| self.lower_abi(abi)),
-                items: self
-                    .arena
-                    .alloc_from_iter(fm.items.iter().map(|x| self.lower_foreign_item_ref(x))),
+                items: self.arena.alloc_from_iter(fm.items.iter().map(|x| {
+                    if fm.safety == Safety::Default {
```

Comment:
is this the right place for this check? We do have an ast validation pass, maybe that's a better fit?

Reply:
This is true, I've done this on lowering thinking that at the right point in ast validation I'd have lost context if this was coming from an extern default block or not but I think this is doable in ast validation. Going to try to move it there.